R/qenv-join.R

```diff
@@ -7,102 +7,102 @@
-#'   match(shared_ids, z@id) # Output: 1 2
-#'   }
-#'   The error occurs because the index position of identical `@id` between the two objects is not the same.
-#'   \item The usage of temporary variable in the code expression could cause `join()` to fail. \cr\cr
-#'   Example:
-#'   \preformatted{
-#'   common_q <- qenv()
-#'   x <- eval_code(
-#'     common_q,
-#'     "x <- numeric(0)
-#'      for (i in 1:2) {
-#'        x <- c(x, i)
-#'      }"
-#'   )
-#'   y <- eval_code(
-#'     common_q,
-#'     "y <- numeric(0)
-#'      for (i in 1:3) {
-#'        y <- c(y, i)
-#'      }"
-#'   )
-#'   q <- join(x,y)
-#'   # Error message will occur
+#' 2. `join()` will look for identical `@id` values in both `qenv` objects.
+#'    The index position of these `@id`s must be the same to determine the evaluation order.
+#'    Otherwise, `join()` will throw an error message.\cr\cr
+#'    Example:
+#'    ```r
+#'    common_q <- qenv(code = "v <- 1", env = list2env(list(v = 1)))
+#'    x <- eval_code(
+#'      common_q,
+#'      "x <- v"
+#'    )
+#'    y <- eval_code(
+#'      common_q,
+#'      "y <- v"
+#'    )
+#'    z <- eval_code(
+#'      y,
+#'      "z <- v"
+#'    )
+#'    q <- join(x, y)
+#'    join_q <- join(q, z)
+#'    # Error message will occur
 #'
-#'   # Check the value of temporary variable i in both objects
-#'   x@env$i # Output: 2
-#'   y@env$i # Output: 3
-#'   }
-#'   `join()` fails to provide a proper result because of the temporary variable `i` exists
-#'   in both objects but has different value.\cr
-#'   To fix this, we can set `i <- NULL` in the code expression for both objects.
-#'   \preformatted{
-#'   common_q <- qenv()
-#'   x <- eval_code(
-#'     common_q,
-#'     "x <- numeric(0)
-#'      for (i in 1:2) {
-#'        x <- c(x, i)
-#'      }
-#'      # dummy i variable to fix it
-#'      i <- NULL"
-#'   )
-#'   y <- eval_code(
-#'     common_q,
-#'     "y <- numeric(0)
-#'      for (i in 1:3) {
-#'        y <- c(y, i)
-#'      }
-#'      # dummy i variable to fix it
-#'      i <- NULL"
-#'   )
-#'   q <- join(x,y)
-#'   }
-#' }
+#'    # Check the order of evaluation based on the id slot
+#'    shared_ids <- intersect(q@id, z@id)
+#'    match(shared_ids, q@id) # Output: 1 3
+#'    match(shared_ids, z@id) # Output: 1 2
+#'    ```
+#'    The error occurs because the index position of identical `@id` between the two objects is not the same.
+#'
+#' 3. The usage of temporary variable in the code expression could cause `join()` to fail.\cr\cr
+#'    Example:
+#'    ```r
+#'    common_q <- qenv()
+#'    x <- eval_code(
+#'      common_q,
+#'      "x <- numeric(0)
+#'       for (i in 1:2) {
+#'         x <- c(x, i)
+#'       }"
+#'    )
+#'    y <- eval_code(
+#'      common_q,
+#'      "y <- numeric(0)
+#'       for (i in 1:3) {
+#'         y <- c(y, i)
+#'       }"
+#'    )
+#'    q <- join(x,y)
+#'    # Error message will occur
+#'
+#'    # Check the value of temporary variable i in both objects\cr
```

Comment:
`\cr` is appearing in the man page as it's inside a code block

```suggestion
#'    # Check the value of temporary variable i in both objects
```